src/ImageSharp/ColorSpaces/Conversion/Implementation/Icc/IccProfileConverter.cs

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Six Labors.
+// Licensed under the Six Labors Split License.
+
+using System.Buffers;
+using System.Numerics;
+using SixLabors.ImageSharp.Advanced;
+using SixLabors.ImageSharp.ColorSpaces.Conversion.Icc;
+using SixLabors.ImageSharp.Memory;
+using SixLabors.ImageSharp.Metadata.Profiles.Icc;
+using SixLabors.ImageSharp.PixelFormats;
+
+namespace SixLabors.ImageSharp.ColorSpaces.Conversion.Implementation.Icc;
+
+/// <summary>
+/// Allows the copnversion between ICC profiles.
+/// </summary>
+internal static class IccProfileConverter
+{
+    /// <summary>
+    /// Performs a conversion of the image pixels based on the input and output ICC profiles.
+    /// </summary>
+    /// <typeparam name="TPixel">The type of pixel.</typeparam>
+    /// <param name="image">The image to convert.</param>
+    /// <param name="inputIccProfile">The input ICC profile.</param>
+    /// <param name="outputIccProfile">The output ICC profile. </param>
+    public static void Convert<TPixel>(Image<TPixel> image, IccProfile inputIccProfile, IccProfile outputIccProfile)
```

Comment:
OK. 

So, my plan with this is to add a new enum to the `DecoderOptions` class called `ColorProfileHandling`.

```c#
/// <summary>
/// Provides enumeration of methods that control how ICC profiles are handled during decode.
/// </summary>
public enum ColorProfileHandling
{
    /// <summary>
    /// Leaves any embedded ICC color profiles intact.
    /// </summary>
    Preserve,

    /// <summary>
    /// Transforms the pixels of the image based on the conversion of any embedded ICC color profiles to sRGB.
    /// The original profile is then removed.
    /// </summary>
    Convert
}
```
We would store a compact sRGB profile created by @saucecontrol from [here](https://github.com/saucecontrol/Compact-ICC-Profiles) as a `ReadOnlySpan<byte>` and if `ColorProfileHandling.Convert` is used, we would pass that as the `outputIccProfile` parameter following decode. This will happen in the `ImageDecoder` class so all our decoders can utilize the approach.

Reply:
This is done now. We default to preserve the original profile for now.